indexes/channels.js

```diff
@@ -0,0 +1,113 @@
+const bipf = require('bipf')
+const pull = require('pull-stream')
+const pl = require('pull-level')
+const jsonCodec = require('flumecodec/json')
+const Plugin = require('ssb-db2/indexes/plugin')
+
+module.exports = function (log, dir) {
+  const bValue = Buffer.from('value')
+  const bContent = Buffer.from('content')
+  const bType = Buffer.from('type')
+  const bChannel = Buffer.from('channel')
+  const bPost = Buffer.from('post')
+
+  let batch = []
+
+  const name = 'channels'
+  const { level, offset, stateLoaded, onData, writeBatch } = Plugin(
+    dir,
+    name,
+    1,
+    handleData,
+    writeData,
+    beforeIndexUpdate
+  )
+
+  function writeData(cb) {
+    level.batch(batch, { keyEncoding: 'json', valueEncoding: 'json' }, cb)
+    batch = []
+  }
+
+  function handleData(record, processed) {
+    if (record.offset < offset.value) return batch.length
+    const recBuffer = record.value
+    if (!recBuffer) return batch.length // deleted
+
+    let p = 0 // note you pass in p!
+    p = bipf.seekKey(recBuffer, p, bValue)
+    if (!~p) return batch.length
+
+    const pContent = bipf.seekKey(recBuffer, p, bContent)
+    if (!~pContent) return batch.length
+
+    const pType = bipf.seekKey(recBuffer, pContent, bType)
+    if (!~pType) return batch.length
+
+    if (bipf.compareString(recBuffer, pType, bPost) === 0) {
+      const content = bipf.decode(recBuffer, pContent)
+      const channel = content.channel
+      if (!channel || channel == '') return batch.length
+
+      updateChannelData(content.channel)
+
+      batch.push({
+        type: 'put',
+        key: channel,
+        value: channels[channel]
+      })
+    }
+
+    return batch.length
+  }
+  
+  function updateChannelData(channel) {
+    if (!channels[channel])
+      channels[channel] = { id: channel, count: 0 }
+    if (!channels[channel].count) // For compatibility with existing index, if it exists.
```

Comment:
I guess this is from your existing index? It's better if you just nuke it so we don't need that code here.